docs/topics/testing/advanced.txt

```diff
@@ -705,6 +705,16 @@ Methods
     Computes and returns a return code based on a test suite, and the result
     from that test suite.
 
+.. method:: DiscoverRunner.log(msg, level=None)
+
+    .. versionadded:: 4.0
+    
+        The ``log`` method was added.
+
+    Logs a message at the given `logging level <https://docs.python.org/3/library/logging.html#levels>`_
```

Comment:
I keep coming back to the docs here. I'm not sure if it is clear enough about what `level` should be (it was obvious once I looked at the tests). The link to the Python docs is great, but it's a massive page to expect folk to understand. I guess in part this ties in with one of the [comments on the ticket](https://code.djangoproject.com/ticket/32552#comment:6) about folk struggling with logging.

Reply:
The Django docs seem to prioritize brevity over completeness, but here is a possible longer version if more guidance is preferred. I would leave it up to the Django maintainers as to how much detail should be included:

> Prints to the console a message with the given integer [logging level](https://docs.python.org/3/library/logging.html#levels) (e.g. `logging.DEBUG`, `logging.INFO`, or `logging.WARNING`), respecting the current `verbosity`. For example, an ``INFO`` message will be logged if the verbosity is at least ``1``, and ``DEBUG`` will be logged if it is at least ``2``.

As an aside, this will make a lot more sense as a documented method once [ticket #32552](https://code.djangoproject.com/ticket/32552) is fully implemented. The reason is that the method is primarily useful for when a `logger` can be configured on the `DiscoverRunner` instance (in which case the user would presumably already be familiar with logging). The behavior of printing to the console will be the fallback behavior in that case, rather than the primary purpose for the method.

Reply:
I like Chris' proposition. I think it makes sense even without a logger in use.